### ISSUE TITLE
OS X 10.10 build error:  'va_start' has undefined behavior with reference types

### ISSUE THREAD

**Issue by jamiebullock** [radarsat1/liblo#20]
Hi!

I'm getting a build errors on OS X 10.10, x86_64

The errors are of the form:

```
../lo/lo_cpp.h:242:25: error: 'va_start' has undefined behavior with reference types [-Werror,-Wvarargs]
```

Perhaps it is better to use C++11 variadic functions instead of C-style variadics?

If you think this is good approach, I'd be happy to submit a pull request with a fix.

$ clang -v
Apple LLVM version 6.0 (clang-600.0.57) (based on LLVM 3.5svn)
Target: x86_64-apple-darwin14.1.0
Thread model: posix

**Comment by radarsat1** [comment]
Hi Jamie, wow I'm surprised this never triggered for me before, but I just compiled with clang on Ubuntu 14.04 and got the same error.  Haven't had time to fix it, but the variadic functions were a bit of a challenge to wrap properly.  Feel free to submit a pull request, otherwise I'll look at it when I get a chance.

**Comment by radarsat1** [comment]
I started looking into variadic templates etc. until i realized this was easily fixed by just not using a reference in the types argument.  No need for it anyways, the string_type class does nothing but encapsulate a pointer.  Fixed in fc6bcb8c51.
